### PR TITLE
🎉 Source Google Directory: support oauth

### LINKED ISSUE #6265: Source Google Direcotry: support oauth
## Tell us about the problem you're trying to solve
With the release of Airbyte Cloud, we need to start supporting Oauth for this connector, since it's the recommended way of authenticating users into a SaaS application. 

If this connector doesn't support oauth already (i.e: doesn't accept a client_id and client_secret) then we need to update its spec to accept those parameters. There are two ways to do this: 

**If the connector already supports some auth mechanism like api_key**, I suggest that this be a oneof nested inside a top-level field called "authentication": 

`
{
  authentication: {
     type: object
     oneOf: [ 
       // api key,
       // oauth
     ] 
  }
}
`

**If the connector only supports webflow oauth**, then no changes are needed to the properties format and we will only need to add annotations. 

See the connector spec reference in the docs for reference on how a oneof can be implemented.

This should be done in a backwards compatible manner i.e: users currently supplying authentication info in the config's top-level should not be impacted by this change. 

## Acceptance Criteria

1. The connector supports oauth webflow authentication with client_id/client_secret in a backwards compatible manner
2. Oauth properties are annotated properly. See [this PR](https://github.com/airbytehq/airbyte/pull/6022) for an example

### PR DESCRIPTION
**Pull request by vitaliizazmic**
## What
Supporting oauth

## How
Source Google Directory already supports authentication via Service account. So oauth webflow support was added. And specification was updated.

Closes #6265
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

### CODE REVIEW

**Review by avida — APPROVED** [review]

**Review by sherifnada — APPROVED** [review]
I didn't scrutinize the implementation, mostly looked at docs/UX. Do we need unit tests for auth?

File: tools/bin/ci_credentials.sh [hunk 1]
```diff
@@ -82,6 +82,7 @@ write_standard_creds source-google-analytics-v4 "$GOOGLE_ANALYTICS_V4_TEST_CREDS
 write_standard_creds source-google-analytics-v4 "$GOOGLE_ANALYTICS_V4_TEST_CREDS_SRV_ACC" "service_config.json"
 write_standard_creds source-google-analytics-v4 "$GOOGLE_ANALYTICS_V4_TEST_CREDS_OLD" "old_config.json"
 write_standard_creds source-google-directory "$GOOGLE_DIRECTORY_TEST_CREDS"
+write_standard_creds source-google-directory "$GOOGLE_DIRECTORY_TEST_CREDS_OAUTH" "config_oauth.json"
```
[the remark this answers is not in the document]
Comment (vitaliizazmic) [reply tools/bin/ci_credentials.sh:85]: @keu SAT isn't enabled in this source. I've created PR #7464. I will add required test there.

**Pull request merged**